book/src/types/objects/index.md

```diff
@@ -0,0 +1,218 @@
+# fn main() {}
+```
+
+If using [Rust doc comments][6] is not desired (for example, when we want to keep [Rust] API docs and GraphQL schema descriptions different), the `#[graphql(description = "...")]` attribute can be used instead, which takes precedence over [Rust doc comments][6]:
+```rust
+# extern crate juniper;
+# use juniper::GraphQLObject;
+#
+/// This doc comment is visible only in Rust API docs.
+#[derive(GraphQLObject)]
+#[graphql(description = "This description is visible only in GraphQL schema.")]
+struct Person {
+    /// This doc comment is visible only in Rust API docs.
+    #[graphql(desc = "This description is visible only in GraphQL schema.")]
+    //        ^^^^ shortcut for a `description` argument
+    name: String,
+
+    /// This doc comment is visible in both Rust API docs and GraphQL schema 
+    /// descriptions.
+    age: i32,
+}
+#
+# fn main() {}
+```
+
+
+### Renaming
+
+By default, [struct] fields are converted from [Rust]'s standard `snake_case` naming convention into [GraphQL]'s `camelCase` convention:
+```rust
+# extern crate juniper;
+# use juniper::GraphQLObject;
+#
+#[derive(GraphQLObject)]
+struct Person {
+    first_name: String, // exposed as `firstName` in GraphQL schema
+    last_name: String,  // exposed as `lastName` in GraphQL schema
+}
+#
+# fn main() {}
+```
+
+We can override the name by using the `#[graphql(name = "...")]` attribute:
+```rust
+# extern crate juniper;
+# use juniper::GraphQLObject;
+#
+#[derive(GraphQLObject)]
+#[graphql(name = "WebPerson")] // now exposed as `WebPerson` in GraphQL schema
+struct Person {
+    name: String,
+    age: i32,
+    #[graphql(name = "websiteURL")]
+    website_url: Option<String>, // now exposed as `websiteURL` in GraphQL schema
+}
+#
+# fn main() {}
+```
+
+Or provide a different renaming policy for all the [struct] fields:
+```rust
+# extern crate juniper;
+# use juniper::GraphQLObject;
+#
+#[derive(GraphQLObject)]
+#[graphql(rename_all = "none")] // disables any renaming
+struct Person {
+    name: String,
+    age: i32,
+    website_url: Option<String>, // exposed as `website_url` in GraphQL schema
+}
+#
+# fn main() {}
+```
+> **TIP**: Supported policies are: `SCREAMING_SNAKE_CASE`, `camelCase` and `none` (disables any renaming).
+
+
+### Deprecation
+
+To [deprecate][9] a [GraphQL object][0] field, the `#[graphql(deprecated = "...")]` attribute should be used:
+```rust
+# extern crate juniper;
+# use juniper::GraphQLObject;
+#
+#[derive(GraphQLObject)]
+struct Person {
+    name: String,
+    age: i32,
+    #[graphql(deprecated = "Please use the `name` field instead.")]
+    first_name: String,
+}
+#
+# fn main() {}
+```
+> **NOTE**: Only [GraphQL object][0]/[interface][11] fields and [GraphQL enum][10] values could be [deprecated][9].
+
+
+### Ignoring
+
+By default, all [struct] fields are included into the generated [GraphQL object][0] type. To prevent including a specific field, it should be annotated with the `#[graphql(ignore)]` attribute:
```

Comment:
```suggestion
By default, all [struct] fields are included into the generated [GraphQL object][0] type. To prevent inclusion of a specific field annotate it with the `#[graphql(ignore)]` attribute:
```